src/main/java/org/zalando/nakadi/repository/KafkaRepositoryCreator.java

```diff
@@ -29,18 +30,25 @@
     private final ZookeeperSettings zookeeperSettings;
     private final KafkaTopicConfigFactory kafkaTopicConfigFactory;
     private final MetricRegistry metricRegistry;
+    private final Integer sessionTimeoutMs;
+    private final Integer connectionTimeoutMs;
 
     @Autowired
-    public KafkaRepositoryCreator(final NakadiSettings nakadiSettings,
-                                  final KafkaSettings kafkaSettings,
-                                  final ZookeeperSettings zookeeperSettings,
-                                  final KafkaTopicConfigFactory kafkaTopicConfigFactory,
-                                  final MetricRegistry metricRegistry) {
+    public KafkaRepositoryCreator(
+            final NakadiSettings nakadiSettings,
+            final KafkaSettings kafkaSettings,
+            final ZookeeperSettings zookeeperSettings,
+            final KafkaTopicConfigFactory kafkaTopicConfigFactory,
+            final MetricRegistry metricRegistry,
+            @Value("${nakadi.zookeeper.exhibitor.session-timeout-ms:10000}") final Integer sessionTimeoutMs,
+            @Value("${nakadi.zookeeper.exhibitor.connection-timeout-ms:3000}") final Integer connectionTimeoutMs) {
```

Comment:
Can you please move the default values to application.yml ?

Reply:
no, it is already quite big, I do not want to move there every parameter of the system

Reply:
The current situation is that you duplicated this in two places

Reply:
ok

Reply:
ahaha https://github.com/zalando/nakadi/blob/master/src/main/resources/application.yml#L75

Reply:
@v-stepanov fixed, thanks